src/index_feature_list_test.cpp

```diff
@@ -0,0 +1,31 @@
+
+// Copyright 2024-present the vsag project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "index_feature_list.h"
+
+#include "catch2/catch_test_macros.hpp"
+
+TEST_CASE("Test set&check", "[ut][index_feature_list]") {
+    using namespace vsag;
+    IndexFeatureList list;
+    auto count = static_cast<uint32_t>(IndexFeature::INDEX_FEATURE_COUNT);
+    for (uint32_t i = 0; i < count; ++i) {
+        REQUIRE(list.CheckFeature(static_cast<IndexFeature>(i)) == false);
+        list.SetFeature(static_cast<IndexFeature>(i));
+        REQUIRE(list.CheckFeature(static_cast<IndexFeature>(i)) == true);
+        list.SetFeature(static_cast<IndexFeature>(i), false);
+        REQUIRE(list.CheckFeature(static_cast<IndexFeature>(i)) == false);
```

Comment:
We need to verify that the current feature's settings do not affect other features.

Reply:
done